sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -98,7 +100,12 @@ case class ExplainCommand(
 
   // Run through the optimizer to generate the physical plan.
   override def run(sparkSession: SparkSession): Seq[Row] = try {
-    val queryExecution = sparkSession.sessionState.executePlan(logicalPlan)
+    val queryExecution =
+      if (logicalPlan.isStreaming) {
+        new IncrementalExecution(sparkSession, logicalPlan, OutputMode.Append(), "dummy", 0)
```

Comment:
Yes. It will apply strategies in `IncrementalExecution`. E.g.,

```
scala> val df = spark.readStream.format("text").load("/Users/zsx/stream")
df: org.apache.spark.sql.DataFrame = [value: string]

scala> df.explain
== Physical Plan ==
StreamingRelation FileSource[/Users/zsx/stream], [value#0]

scala> val df2 = df.groupBy("value").count()
df2: org.apache.spark.sql.DataFrame = [value: string, count: bigint]

scala> df2.explain
== Physical Plan ==
*HashAggregate(keys=[value#0], functions=[count(1)])
+- StateStoreSave [value#0], OperatorStateId(dummy,0,0), false
   +- *HashAggregate(keys=[value#0], functions=[merge_count(1)])
      +- StateStoreRestore [value#0], OperatorStateId(dummy,0,0)
         +- *HashAggregate(keys=[value#0], functions=[merge_count(1)])
            +- Exchange hashpartitioning(value#0, 200)
               +- *HashAggregate(keys=[value#0], functions=[partial_count(1)])
                  +- StreamingRelation FileSource[/Users/zsx/stream], [value#0]
```